pymie/mie.py

```diff
@@ -646,41 +738,87 @@ def integrate_intensity_complex_medium(I_par, I_perp, distance, theta, k,
     
     Parameters
     ----------
-    I_par, I_perp: differential scattered intensities
-    distance: distance away from the scatterer (Quantity in [length])
-    theta: array scattering angles (Quantity in rad)
+    I_1, I_2: nd arrays 
+        differential scattered intensities, can be functions of theta or of
+        theta and phi
+    distance: float (Quantity in [length])
+        distance away from the scatterer 
+    thetas: nd array (Quantity in rad)
+        scattering angles
     k: wavevector given by 2 * pi * n_medium / wavelength 
        (Quantity in [1/length])
-    phi_min: minimum azimuthal angle, default set to 0 (Quantity in rad).
-    phi_max: maximum azimuthal angle, default set to 2pi (Quantity in rad).
+    phi_min: float (Quantity in rad).
+        minimum azimuthal angle, default set to 0 
+        optional, only necessary if coordinate_system is 'scattering plane'
+    phi_max: float (Quantity in rad).
+        maximum azimuthal angle, default set to 2pi 
+        optional, only necessary if coordinate_system is 'scattering plane'
+    phis: None or ndarray
+        azimuthal angles
     
     Returns
     -------
-    sigma: integrated cross section (in units of length**2)
- 
+    sigma: float (in units of length**2)
+        integrated cross section 
+    sigma_1: float (in units of length**2)
+        integrated cross section for first component of basis
+    sigma_2: float (in units of length**2)
+        integrated cross section for second component of basis
+    dsigma_1: ndarray (in units of length**2)
+        differential cross section for first component of basis
+    dsigma_2: ndarray (in units of length**2)
+        differential cross section for second component of basis
```

Comment:
I'm a little confused about why this function returns a differential scattering cross section.  The docs (and function name) say it returns the integral.

Reply:
This function has always returned the differential scattering cross section in addition to the integrated cross section. This was originally written this way for testing purposes and should probably now be removed. However, since montecarlo.py relies on this function and expects a tuple of length 5, I think it would be better to change this as a part of the refactoring. I can raise an issue so we don't forget to fix this.